### ISSUE TITLE
Missing license database: /usr/share/rpminspect/licenses//usr/share/fedora-license-data/licenses/fedora-licenses.json: No such file or directory

### ISSUE THREAD

**Issue by hroncok** [rpminspect/rpminspect#858]
```
$ rpm -qa | grep -E 'rpminspect|license-data'
fedora-license-data-1.3-1.20220825152831108795.main.0.g32b8e63.fc35.noarch
librpminspect-1.10-1.fc35.x86_64
rpminspect-1.10-1.fc35.x86_64
rpminspect-data-fedora-1.9-1.fc35.noarch

$ rpminspect-fedora -T license .../python3.11-3.11.0~rc1-2.fc38.src.rpm 
diagnostics:
------------
1) Version information for libraries and programs used by rpminspect.  This result is for informational and diagnostic purposes only. 

Result: INFO
Waiver Authorization: Not Waivable

Details:
rpminspect version 1.9
zlib version 1.2.11
libmagic version 540
libclamav version 0.103.7
librpm version 4.17.1
libxml version 20914
json-c version 0.15
libcurl version 7.79.1 (OpenSSL/1.1.1q-fips zlib/1.2.11 brotli/1.0.9 libidn2/2.3.3 libpsl/0.21.1 (+libidn2/2.3.2) libssh/0.9.6/openssl/zlib nghttp2/1.45.1 OpenLDAP/2.4.59)
libarchive version 3.5.3 (zlib/1.2.11 liblzma/5.2.5 bz2lib/1.0.8 liblz4/1.9.3 libzstd/1.5.2)
libyaml version 0.2.5
OpenSSL version 1.1.1q  FIPS 5 Jul 2022
xmlrpc-c version 1.51.0
msgunfmt (GNU gettext-tools) 0.21
diffstat version 1.64
annocheck version 10.79.
abidiff version 2.0.0
kmidiff version 2.0.0


2) Command line arguments used to invoke rpminspect. 

Result: INFO
Waiver Authorization: Not Waivable

Details:
/usr/bin/rpminspect -c /usr/share/rpminspect/fedora.yaml -T license .../python3.11-3.11.0~rc1-2.fc38.src.rpm



license:
--------
1) Missing license database: /usr/share/rpminspect/licenses//usr/share/fedora-license-data/licenses/fedora-licenses.json: No such file or directory 

Result: BAD
Waiver Authorization: Not Waivable

Suggested Remedy: Make sure the licensedb setting in the rpminspect configuration is set to a valid licensedb file.  This is also commonly due to a missing vendor specific rpminspect-data package on the system.

```

The relevant part from `/usr/share/rpminspect/fedora.yaml` is:

```yaml
    # Either the name of a license database file under the 'licenses/'
    # subdirectory in the vendor_data_dir or a full path to a license
    # database file to use.  This file is used by the 'license'
    # inspection.
    licensedb: /usr/share/fedora-license-data/licenses/fedora-licenses.json
```

Apparently, this is appended to the `/usr/share/rpminspect/licenses/` string instead of recognized as an absolute path. When I patch it to `../../fedora-license-data/licenses/fedora-licenses.json` and when I create the `/usr/share/rpminspect/licenses` directory (which is missing otherwise), it works.

**Comment by dcantrell** [comment]
I can't reproduce this locally.  Here's where it determines where or not to prepend the data subdirectory:

```
    if (ri->licensedb && ri->licensedb[0] == '/') {
        actual_licensedb = strdup(ri->licensedb);
    } else {
        xasprintf(&actual_licensedb, "%s/%s/%s", ri->vendor_data_dir, LICENSES_DIR, ri->licensedb);
    }
```

That's currently how things are in the latest builds and released versions.  Parse error maybe on fedora.yaml?  Do you have a local rpminspect.yaml file?

**Comment by hroncok** [comment]
no

**Comment by hroncok** [comment]
```
$ podman run --rm -ti fedora:35 /usr/bin/bash
[root@bf3f4c5d8ece /]# dnf install rpminspect rpminspect-data-fedora wget
...
[root@bf3f4c5d8ece /]# wget https://kojipkgs.fedoraproject.org//work/tasks/165/91170165/python3.11-3.11.0~rc1-2.fc38.src.rpm
...
[root@bf3f4c5d8ece /]# rpminspect-fedora -T license ./python3.11-3.11.0~rc1-2.fc38.src.rpm 
...
1) Missing license database: /usr/share/rpminspect/licenses//usr/share/fedora-license-data/licenses/fedora-licenses.json: No such file or directory 
...
```

**Comment by hroncok** [comment]
Same with `fedora:36`

**Comment by dcantrell** [comment]
Ahhh, ok, I see.  This bug is present in rpminspect-1.10, but I have already fixed it in the main branch.  The fix will be in the 1.11 release.  For now you can install rpminspect builds from my Copr repo to get the fix.

**Comment by dcantrell** [comment]
Fixed in commit 8c59d8dd0756692194937ca2a087d45ee564b5ba on 06-Jul-2022.

**Comment by hroncok** [comment]
Indeed. Thanks.
